src/anomalib/models/efficient_ad/torch_model.py

```diff
@@ -24,6 +24,16 @@ def imagenet_norm_batch(x):
     return x_norm
 
 
+def reduce_tensor_elems(tensor: torch.Tensor, device, max_input_size=2**24) -> torch.Tensor:
+    tensor = torch.flatten(tensor)
+    if len(tensor) > max_input_size:
+        # select a random subset with max_input_size elements.
+        perm = torch.randperm(len(tensor), device=device)
```

Comment:
As I mentioned below, instead of passing `device` as a function argument, we could get the device from the tensor itself. Something like 

```suggestion
        perm = torch.randperm(len(tensor), device=tensor.device)
```